### ISSUE TITLE
Bug coming from the outdated System.Data.SqlClient dependency

### ISSUE THREAD

**Issue by YeskaNova** [fluentmigrator/fluentmigrator#1417]
Hi,

I use FluentMigrator in my app to check if a SQL table exists. My whole app crashes when I stress-test it. I have narrowed the issue to this crazy-to-debug bug: https://github.com/dotnet/runtime/issues/19735 . FluentMigrator for SqlServer uses System.Data.SqlClient version 4.4.3 . This bug has been fixed in 4.5.0-RC1. Can we update that dependency to anything more recent than that? ( If we can update it to Microsoft.Data.SqlClient that would be even better.)

**Comment by jzabroski** [comment]
Immediate workaround would be to force your library to handle AssemblyResolve.

Build is currently broken due to AzureDevOps StackOverflowException.  Not sure on ETA to fix it.

**Comment by YeskaNova** [comment]
Yes, AssemblyResolve can do the trick, I'll do that for the time being. It took me weeks to figure out where was the error so if we can put this for the next release that would save time for someone else in the future. Thank you.
